db/external_sst_file_test.cc

```diff
@@ -68,24 +70,24 @@ class ExternSSTFileLinkFailFallbackTest
     : public ExternalSSTFileTestBase,
       public ::testing::WithParamInterface<std::tuple<bool, bool>> {
  public:
-  ExternSSTFileLinkFailFallbackTest()
-      : test_env_(new ExternalSSTTestEnv(env_, true)) {
+  ExternSSTFileLinkFailFallbackTest() {
+    fs_ = std::make_shared<ExternalSSTTestFS>(env_->GetFileSystem(), true);
+    test_env_ = NewCompositeEnv(fs_);
```

Comment:
I suggest `test_env_.reset(new CompositeEnvWrapper(env_, fs_))` instead, in case `env_` has overridden any non-storage APIs.